### DISCUSSION TITLE
What does 'tokio' mean?

### DISCUSSION THREAD

**Discussion by sequix** [tokio-rs/tokio#3668] [General]
Is `tokio` the translation of japaness `時お` (Time! in a exclamation way...), or the capital of Japan `Tokyo` ?

And why the logo of tokio is a spider-like eye?

**Comment by Darksonn** [comment]
The name was chosen because it ends in `io` and is fairly unique. Beyond that, Tokyo is a nice place to visit, but uniqueness and `io` are the main reasons.
I don't know where the logo comes from.

**Reply by memoryruins** [reply]
I've pictured the logo as also playing on `io`, where `i`s encircle an `o` in the center. Only a guess though.

**Reply by hawkw** [reply]
> The name was chosen because it ends in `io` and is fairly unique. Beyond that, Tokyo is a nice place to visit, but uniqueness and `io` are the main reasons.
> I don't know where the logo comes from.

I have a _vague_ memory of someone saying that part of the motivation for the name came from a contraction of "TOKen IO", to reference [`mio`'s `Token` type](https://docs.rs/mio/0.7.11/mio/struct.Token.html); this dates back to the early days of the project when the  connection with `mio` was more user-facing, as well as being a pun on "Tokyo". I'm not sure if this etymology is correct, but it's appealing to me. :)

As for the logo, I interpret it as a stylized representation of an event loop, with the spokes as various event sources like timers and IO resources. To me, it also kind of suggests some kind of science-fictional "reactor core"...if you were there for some of the earliest versions of Tokio, you may remember the [`reactor::Core`](https://docs.rs/tokio-core/0.1.18/tokio_core/reactor/struct.Core.html) type, which was once a major part of the API in the _very_ early days of the project. But that's just my reading, I'm not sure of the entire story behind it...

**Reply by memoryruins** [reply]
Found an old [reddit comment](https://www.reddit.com/r/rust/comments/4vzomj/announcing_tokio_a_finagle_inspired_network/d62sq9a/) that asked about the name and Carl responded with "io" / "tokyo", while another commenter suggests "token + io".

@hawkw 

> To me, it also kind of suggests some kind of science-fictional "reactor core"

I like that ^^. It's reminiscent of Dr. Manhattan's choice of a [hydrogen symbol](https://en.wikipedia.org/wiki/Doctor_Manhattan#/media/File:Dr_Manhattan_symbol.svg) in Watchmen.

**Reply by sequix** [reply]
Still not good at rust through, knowing something like this keeps me going. 

Yes, I live on programmer meme... : ）

**Comment by Jimmy-Z** [comment]
I thought the author is a fan of Tokio, a Japanese band 😆

**Comment by Dzzt** [comment]
The logo is obviously inspired by 
https://en.wikipedia.org/wiki/Tokyo#/media/File:Emblem_of_Tokyo_Metropolis.svg
This mark is the combined figure of stylized 4 kanji letters 日本東京(Japan, Tokyo).
